modules/oe_content_persistent/src/Controller/PersistentUrlController.php

```diff
@@ -60,8 +73,13 @@ public function index(string $uuid): RedirectResponse {
       // \Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber.
       // More information you could find in this article:
       // https://www.lullabot.com/articles/early-rendering-a-lesson-in-debugging-drupal-8
-      if ($entity instanceof TranslatableInterface) {
-        return new RedirectResponse($entity->toUrl()->toString(), 302, ['PURL' => '1']);
+      if ($entity instanceof EntityInterface) {
+        // Not all entity types will need to be linked to their canonical URLs,
+        // so we dispatch an event to allow to modify the resulting URL.
+        $event = new PersistentUrlResolverEvent($entity);
```

Comment:
Never good idea to dispatch the same event in two different places. So let's make a service that is responsible for resolving the link of a given entity. In that, we default to canonical and we dispatch the event. And we use that service here and in the filter.